app/src/test/webapp/app/shared/esbmsgsvc.test.js

```diff
@@ -0,0 +1,34 @@
+describe("EsbMessageService", function() {
+  var EsbMessageService;
+  var $httpBackend;
+
+  var resubmitRequest = {
+    "id": 1,
+    "payload": "<foo></foo>"
+  };
+
+  var resubmitResponse = {
+    "status": "Status",
+    "data": {
+      "field": "Present"
+    }
+  };
+
+  beforeEach(module("esbMessageAdminApp"));
+
+  beforeEach(inject(function(_$httpBackend_, _EsbMessageService_){
+    $httpBackend = _$httpBackend_;
+    EsbMessageService = _EsbMessageService_;
+  }));
+
+  describe("after calling resubmit", function(){
+
+    it("should return a promise with the server response", function() {
+      $httpBackend.expectPOST("api/key/resubmit/1", resubmitRequest).respond(200, resubmitResponse);
+      EsbMessageService.resubmitMessage(resubmitRequest).then(function(response) {
+        expect(response).toEqual(resubmitResponse);
+      });
```

Comment:
Have you tested this test fails if the service is broken? Looking at this I think it does not actually test anything because you're not flushing the `$httpBackend`

When writing any test, you should always see it fail, whether that's because the feature is implemented yet, or after the fact you purposely break the feature, you "test your test" actually fails when it's supposed to. In other words, a good test is designed to fail :). Not saying you haven't done this, just kind of ranting just in case.